graylog2-server/src/main/java/org/graylog/security/authservice/ldap/LDAPUser.java

```diff
@@ -22,6 +22,8 @@
 public abstract class LDAPUser {
     public abstract String base64UniqueId();
 
+    public abstract ADUserAccountControl userAccountControl();
```

Comment:
I don't think we should put AD specific code into the `LDAPEntry` and `LDAPUser` objects. How about using the `AccountStatus` instead? That makes it independent from AD. We could extract the `AccountStatus` into a separate enum out of the `User` interface to make it a bit more generic.
What do you think?

I also think we should only have an `accountStatus()` method in `LDAPUser` instead of also putting it into the `LDAPEntry` object. The `LDAPEntry` object is supposed to be generic. We can move the `userAccountControl` parsing into the `createLDAPUser()` method in `UnboundLDAPConnector`.

Reply:
I could remove it from `LDAPEntry`, because it would be still accessible within the attributes map.
But I'm not sure about using `AccountStatus` for that. A simple enum cannot reflect the `userAccountControl` flags.

Reply:
Ah, right. But do we need all the flags? We are only interested in the the account status here, no?

Reply:
If we want to display them in the test result.
I don't see any other way right now :-/

Reply:
Okay, if it's just for the test result I am fine with leaving the flags out. :smile: We can still show if a user is enabled/disabled in the test result.

Reply:
ok. :wastebasket: